addons/sale_timesheet/report/timesheets_analysis_report.py

```diff
@@ -36,11 +36,13 @@ def _select(self):
             A.so_line AS so_line,
             A.timesheet_invoice_type AS timesheet_invoice_type,
             A.timesheet_invoice_id AS timesheet_invoice_id,
-            CASE WHEN A.order_id IS NULL THEN 0 ELSE A.unit_amount * SOL.price_unit END AS timesheet_revenues,
+            CASE WHEN A.order_id IS NULL THEN 0 ELSE A.unit_amount * SOL.price_unit * sol_product_uom.factor / a_product_uom.factor END AS timesheet_revenues,
             CASE WHEN A.order_id IS NULL THEN 0 ELSE A.unit_amount END AS billable_time
         """
 
     @api.model
     def _from(self):
         return super()._from() + """
-        LEFT JOIN sale_order_line SOL ON A.so_line = SOL.id"""
+        LEFT JOIN sale_order_line SOL ON A.so_line = SOL.id
+        LEFT JOIN uom_uom sol_product_uom ON sol_product_uom.id=SOL.product_uom
+        LEFT JOIN uom_uom a_product_uom ON a_product_uom.id=A.product_uom_id"""
```

Comment:
I am wondering if we should not do an `INNER JOIN` since `product_uom_id` should be set for each timesheet and so we avoid having null value in `a_product_uom.factor` column. 🤔 
```suggestion
        INNER JOIN uom_uom a_product_uom ON a_product_uom.id=A.product_uom_id"""
```

What do you think? 🤔